torch/csrc/autograd/saved_variable.cpp

```diff
@@ -100,6 +105,11 @@ Variable SavedVariable::unpack(std::shared_ptr<Node> saved_for) const {
     throw std::logic_error("No grad accumulator for a saved leaf!");
   impl::set_grad_accumulator(var, grad_accumulator_);
 
+  if (fw_grad_ && !fw_grad_->empty()) {
+    auto new_fw_grad = fw_grad_->value(/* level */ 0);
+    var.set_fw_grad(new_fw_grad, /* level */ 0, /* is_inplace_op */ false);
```

Comment:
How does this interact with views?

Reply:
The answer is, var is never a view (because we never do view recovery) and so we're done.